release.rb

```diff
@@ -0,0 +1,215 @@
+      unless updated_podspec_version 
+        restore_podspec
+        die("Failed to update version in podspec. Aborting.")
+      end
+
+      added_podspec = system("git add #{pod_spec}")
+      unless added_podspec
+        restore_podspec
+        die("Failed to add #{pod_spec} to index. Aborting.")
+      end
+
+      pushed_updated_version = system("git commit -m \"Bumping version to #{version}\"")
+      unless pushed_updated_version
+        restore_podspec
+        die("Failed to push updated version. Aborting.")
+      end
+    else
+      puts " > Podspec already set to #{version}. Skipping."
+    end
+
+    if force
+      puts "-> Tagging version"
+      tagged_version = system("git tag -s #{version_tag} -F #{release_notes}")
+      
+      die("Failed to tag version. Aborting.") unless tagged_version
+      puts "-> Pushing tag to origin"
+      pushed_tag = system("git push origin #{version_tag}")
+      die("Failed to push tag '#{version_tag}' to origin") unless pushed_tag
+    else
+       puts "-> Tagging version (force)"
+       tagged_version = system("git tag -f -s #{version_tag} -F #{release_notes}")
+       die("Failed to tag version. Aborting.") unless tagged_version
+
+       pushed_tag = system("git push origin #{version_tag}")
+       die("Failed to push tag #{version_tag} to origin. Aborting.") unless pushed_tag
+    end
+
+    if set_podspec_version
+      `rm #{release_notes}`
+      pushed_to_origin = system("git push origin #{remote_branch}")
+      die("Failed to push to origin. Aborting.") unless pushed_to_origin
+
+      puts " > Pushed version to origin"
+    end
+
+    puts
+    puts "---------------- Released as $VERSION_TAG ----------------"  
+    puts
+    puts "Archiving carthage release..."
+    
+    archived_carthage_release = system("#{carthage} archive #{carthage_framework_name}")
+    die("Failed to archive framework for carthage. Aborting.") unless archived_carthage_release
+
+    puts
+    #puts "Pushing to pod trunk..."
+
+    #`#{pod} trunk push #{pod_spec}`
+
+    puts
+    puts "================ Finalizing the Release ================"
+    puts
+    puts " - Go to #{github_tags_url} and mark this as a release."
+    puts "   - Paste the contents of #{release_notes} into the release notes. Tweak for Github styling."
+    puts "   - Attach #{carthage_framework_name}.framework.zip to it."
+    puts " - Announce!"
+
+    `rm #{pod_spec}.backup`
+  end
+
+  def self.die(error_message)
+    raise("[ERROR]: #{error_message}")
+  end
+
+  def self.verify_availability(command) 
+    system("which #{command}")
+  end
+
+  def self.verify_version(version, force) 
+    die("This tag (#{version}) is an incorrect format. You should remove the 'v' prefix.") unless /^[vv]/.match(version)
+    die("This tag (#{version}) is an incorrect format. It should be in 'v{MAJOR}.{MINOR}.{PATCH}(-{PRERELEASE_NAME}.{PRERELEASE_VERSION})' form.") unless /^v(\d+\.)(\d+\.)(\d+)(-\w+(\.\d+)?)?$/.match(version)
```

Comment:
Your regex seems absolutely correct to me! 💯 

My two cents: we could simplify this regex and our lives by just dumping the concept of a "prerelease version". I know Nimble has done prereleases in the past ([v2.0.0-rc.2](https://github.com/Quick/Nimble/releases/tag/v2.0.0-rc.2)), but personally I think Quick org projects aren't big/important enough to warrant this. If we push out a bad v1.0, we'll release a v2.0 -- not the end of the world.

Thoughts, @Quick/core?